### PR TITLE
Fix issue with OV enrolment with only high security methods enabled for mobile devices

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## Description:

Issue described in https://oktainc.atlassian.net/browse/OKTA-745318

See videos below.

Can be locally reproduced with mock `authenticator-enroll-ov-same-device` and enabling mobile browser emulation.

PR adds 2 Bacon tasks (`parity-v3-mobile` and `testcafe-mobile`) to test spec `test/testcafe/spec/EnrollAuthenticatorOktaVerify_spec.js` in Chrome with mobile emulation

## Bug description

This line forces dialog to select OV enrollment channel for mobile devices:

https://github.com/okta/okta-signin-widget/blob/d314a99c3b0344f94e0144185404d004d2b60a57/src/v3/src/components/Widget/index.tsx#L296-L299

But if the only one channel available is “SAMEDEVICE“ and it is already selected (which happens if “Higher security methods“ is selected in `Security -> Authenticators  -> Okta Verify` as on screenshot 1), then in this file

https://github.com/okta/okta-signin-widget/blob/d314a99c3b0344f94e0144185404d004d2b60a57/src/v3/src/transformer/oktaVerify/transformOktaVerifyChannelSelection.ts#L68-L69

```js
channelSelectionElement.options.customOptions == []
```

which is unexpected by this line

https://github.com/okta/okta-signin-widget/blob/d314a99c3b0344f94e0144185404d004d2b60a57/src/v3/src/transformer/oktaVerify/transformOktaVerifyChannelSelection.ts#L76

and causes runtime error 
```
TypeError: Cannot read properties of undefined (reading 'value')
```

Bug fix:
https://github.com/okta/okta-signin-widget/blob/d314a99c3b0344f94e0144185404d004d2b60a57/src/v2/models/AppState.ts#L282

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-745318](https://oktainc.atlassian.net/browse/OKTA-745318)

### Screenshot/Video:

Screenshot 2024-07-25 at 14 06 53

Bug reproduction in Gen3:

https://github.com/user-attachments/assets/c961e64d-8d4f-4192-b817-3e8f072b32dc

Expected UI after click "Set up":

Screenshot 2024-07-25 at 20 18 15

Bug is not present in Gen2:

https://github.com/user-attachments/assets/b76e24a4-2204-413c-9722-341f25fb43ad

### CODE REVIEW

File: .testcaferc.js [hunk 1]
```diff
@@ -61,17 +64,28 @@ const env = {
...
 
+const chromeName = env.CHROME_HEADLESS ? 'chrome:headless' : 'chrome';
+const chromeOptions = env.OKTA_SIW_MOBILE ? ':emulation:device=iphone X' : '';
+const chromeFlags = env.CHROME_HEADLESS ? '' : ' --disable-search-engine-choice-screen';
+const chromeFullName = `${chromeName}${chromeOptions}${chromeFlags}`;
+
 const config = {
-  browsers: [ 'chrome:headless' ],
+  browsers: [chromeFullName],
+  hostname: env.CI ? undefined : 'localhost',
```
Comment (denysoblohin-okta) [inline .testcaferc.js:80]: Without using `--hostname= localhost` locally there can be issue with proxy
```
ERROR Cannot establish one or more browser connections.

Hints:
- The host machine may not be powerful enough to handle the specified concurrency factor (4). Try to decrease the concurrency factor or allocate more computing resources to the host machine.
- Increase the value of the "browserInitTimeout" option if it is too low (currently: 2 minutes for local browsers and 6 minutes for remote browsers). This option determines how long TestCafe waits for browsers to be ready.
- The error can also be caused by network issues or remote device failure. Make sure that your network connection is stable and you can reach the remote device.

Type "testcafe -h" for help.
```

File: package.json [hunk 2]
```diff
@@ -76,12 +76,14 @@
     "test:e2e": "yarn workspace @okta/e2e test",
     "test:e2e:lang": "yarn workspace @okta/e2e test:lang",
     "test:testcafe-setup": "mkdir -p test-reports/testcafe && yarn start --mock mockDuo --open false",
-    "test:testcafe-run": "yarn test -t testcafe -c 4 chrome:headless --reporter spec,xunit:build2/reports/junit/testcafe-results.xml",
+    "test:testcafe-run": "yarn test -t testcafe --reporter spec,xunit:build2/reports/junit/testcafe-results.xml",
```
Comment (denysoblohin-okta) [inline package.json:79]: Moved concurrency and browser settings from CLI to `.testcaferc.js` to be able to configure them depending on environment

File: .testcaferc.js [hunk 3]
```diff
@@ -61,17 +64,28 @@ const env = {
...
+  browsers: [chromeFullName],
+  hostname: env.CHROME_HEADLESS ? undefined : 'localhost',
   clientScripts: env.OKTA_SIW_EN_LEAKS ? [
   ] : [
     { module: 'axe-core/axe.min.js' },
     { module: '@testing-library/dom/dist/@testing-library/dom.umd.js' }
   ],
-  src: env.OKTA_SIW_EN_LEAKS ? [
+  src: env.OKTA_SIW_MOBILE ? [
+    'test/testcafe/spec/EnrollAuthenticatorOktaVerify_spec.js',
```
Comment (denysoblohin-okta) [inline .testcaferc.js:87]: It's the only spec depending on mobile/desktop browser type

**Review by shuowu-okta — APPROVED** [review]

File: .testcaferc.js [hunk 4]
```diff
@@ -95,7 +110,7 @@ const config = {
   }),
 
   // limit concurrency when running flaky tests
-  concurrency: OKTA_SIW_ONLY_FLAKY ? 1 : undefined,
+  concurrency: OKTA_SIW_ONLY_FLAKY || !env.CHROME_HEADLESS ? 1 : (env.OKTA_SIW_EN_LEAKS ? 2 : 4),
```
Comment (lesterchoi-okta) [inline .testcaferc.js:113]: little confusing to read.
```suggestion
  // OKTA_SIW_ONLY_FLAKY: 1 ← flaky
  // !CHROME_HEADLESS: 1 ← for debugging tests
  // OKTA_SIW_EN_LEAKS: 2 ← somewhat flaky
  // !OKTA_SIW_EN_LEAKS: 4 ← most tests
  concurrency: (OKTA_SIW_ONLY_FLAKY || !env.CHROME_HEADLESS) 
    ? 1 
    : (env.OKTA_SIW_EN_LEAKS ? 2 : 4),
```

File: .testcaferc.js [hunk 5]
```diff
@@ -61,17 +64,28 @@ const env = {
...
+  OKTA_SIW_MOBILE: OKTA_SIW_MOBILE === 'true',
+  CHROME_HEADLESS: (CHROME_HEADLESS === 'true' || CI === 'true'),
+  CI: CI === 'true',
   UPDATE_SCREENSHOTS: UPDATE_SCREENSHOTS === 'true',
 };
 
+const chromeName = env.CHROME_HEADLESS ? 'chrome:headless' : 'chrome';
+const chromeOptions = env.OKTA_SIW_MOBILE ? ':emulation:device=iphone X' : '';
+const chromeFlags = env.CHROME_HEADLESS ? '' : ' --disable-search-engine-choice-screen';
+const chromeFullName = `${chromeName}${chromeOptions}${chromeFlags}`;
```
Comment (lesterchoi-okta) [inline .testcaferc.js:76]: 👍

**Review by lesterchoi-okta — APPROVED** [review]

**Review by lesterchoi-okta — APPROVED** [review]

**Pull request merged**